### PR TITLE
Feature/affine gaps

### PR DESCRIPTION
**Pull request by joergi-w**
Adds the affine gaps configuration element and fixes an issue in configuration_fn_proxy. 
Solves issue #323.

### CODE REVIEW

**Comment by rrahn** [comment]
You will need to rebase it once #295 is merged.

**Comment by rrahn** [comment]
We also need to wait for the gap_scheme.

**Comment by rrahn** [comment]
@joergi-w could you please rebase the gaps adaptor to the gap scheme #371.

File: test/unit/alignment/configuration/align_config_gap_test.cpp [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+    {
+        detail::configuration cfg = align_cfg::gap(gap_score{-2.2f}, gap_open_score{-4.1f});
+
+        EXPECT_FLOAT_EQ(get<0>(cfg).value.get_gap_score(), -2.2f);
+        EXPECT_FLOAT_EQ(get<0>(cfg).value.get_gap_open_score(), -4.1f);
+        EXPECT_TRUE((std::is_same_v<remove_cvref_t<decltype(cfg)>,
+                                    detail::configuration<detail::align_config_gap<gap_scheme<float>>>>));
+    }
+}
```
Comment (rrahn) [inline test/unit/alignment/configuration/align_config_gap_test.cpp]: Can you also add a test for directly invoking with gap_scheme

Comment (joergi-w) [reply test/unit/alignment/configuration/align_config_gap_test.cpp]: Sure, I added tests for gap_scheme and fixed the doc error.
I rebased on top of the master branch for the Jenkins performance test.

**Comment by rrahn** [comment]
let's merge it

**Pull request merged**